src/main/java/dev/j3fftw/litexpansion/machine/WaterSprinkler.java

```diff
@@ -0,0 +1,152 @@
+import me.mrCookieSlime.Slimefun.cscorelib2.item.CustomItem;
+
+/**
+ * The {@link WaterSprinkler} speeds up the growth of nearby crops
+ * when water is under the machine
+ * Essentially a modified {@link CropGrowthAccelerator}
+ *
+ * @author FluffyBear
+ */
+public class WaterSprinkler extends AbstractGrowthAccelerator {
+
+    private final Set<Material> crops = EnumSet.noneOf(Material.class);
+    public static final int ENERGY_CONSUMPTION = 16;
+    public static final int CAPACITY = 64;
+    private static final int RADIUS = 2;
+    private static final int PROGRESS_SLOT = 4;
+    private static final CustomItem noWaterItem = new CustomItem(Material.BUCKET,
+        "&cNo water found",
+        "",
+        "&cPlease place water under the sprinkler!"
+    );
+    private static final CustomItem waterFoundItem = new CustomItem(Material.WATER_BUCKET,
+        "&bWater detected"
+    );
+
+    public WaterSprinkler() {
+        super(Items.LITEXPANSION, Items.WATER_SPRINKER, RecipeType.ENHANCED_CRAFTING_TABLE,
+            new ItemStack[] {
+                Items.REFINED_IRON, SlimefunItems.ELECTRIC_MOTOR, Items.REFINED_IRON,
+                new ItemStack(Material.BUCKET), Items.MACHINE_BLOCK, new ItemStack(Material.BUCKET),
+                Items.REFINED_IRON, SlimefunItems.MEDIUM_CAPACITOR, Items.REFINED_IRON
+            });
+
+        createPreset(this, Items.WATER_SPRINKER.getImmutableMeta().getDisplayName().orElse("&bWater Sprinkler"),
+            blockMenuPreset -> {
+                for (int i = 0; i < 9; i++)
+                    blockMenuPreset.addItem(i, ChestMenuUtils.getBackground(), ChestMenuUtils.getEmptyClickHandler());
+
+                blockMenuPreset.addItem(PROGRESS_SLOT, noWaterItem);
+            });
+
+        crops.add(Material.WHEAT);
+        crops.add(Material.POTATOES);
+        crops.add(Material.CARROTS);
+        crops.add(Material.NETHER_WART);
+        crops.add(Material.BEETROOTS);
+        crops.add(Material.COCOA);
+        crops.add(Material.SUGAR_CANE);
+
+        if (SlimefunPlugin.getMinecraftVersion().isAtLeast(MinecraftVersion.MINECRAFT_1_14)) {
+            crops.add(Material.SWEET_BERRY_BUSH);
+        }
+    }
+
+    public int getEnergyConsumption() {
+        return ENERGY_CONSUMPTION;
+    }
+
+    @Override
+    public int getCapacity() {
+        return CAPACITY;
+    }
+
+    public int getRadius() {
+        return RADIUS;
+    }
+
+    public int[] getInputSlots() {
+        return new int[0];
+    }
+
+    public int[] getOutputSlots() {
+        return new int[0];
+    }
+
+    @Override
+    protected void tick(Block b) {
+        BlockMenu inv = BlockStorage.getInventory(b);
+
+        if (b.getRelative(BlockFace.DOWN).getType() == Material.WATER) {
+            inv.replaceExistingItem(PROGRESS_SLOT, waterFoundItem);
+        } else {
+            inv.replaceExistingItem(PROGRESS_SLOT, noWaterItem);
+            return;
+        }
```

Comment:
don't do this if the inventory isn't even open it's just useless processing then